### PR TITLE
chore: prevent gh workflows with secrets on forks

### PR DESCRIPTION
**Pull request by bradleystachurski**
Fixes https://github.com/fedimint/fedimint/issues/3129

Followup to https://github.com/fedimint/fedimint/pull/3114#pullrequestreview-1612985083

### CODE REVIEW

File: .github/workflows/ci-nix.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ on:
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   lint:
+    if: github.repository == 'fedimint/fedimint'
```
Comment (elsirion) [inline .github/workflows/ci-nix.yml:23]: I think this should work on forks if we only set the `authToken` if it's available and gives at least rudimentary feedback

**Review by elsirion — COMMENTED** [review]
There are two reasons for why a build might not work:
* Secrets being unavailable
  * Often this should be fixable by running the cachix action read-only
* Runners being unavailable because we use BuildJet and the fork isn't
  * We could supply alternative config, but GH runners are so slow that it's probably best to just run CI locally (easily possible due to nix) or by just opening a draft PR.

File: .github/workflows/ci-nix.yml (hunk 1 above)
Comment (dpc) [reply .github/workflows/ci-nix.yml:23]: What doesn't work about lints on forks? cachix uses `continue-on-error: true`

**Review by dpc — COMMENTED** [review]
Seems like `cachix` should not be a blocker for forks. They can still potentially fetch stuff from public fedimint's cachix, and not being able to push is ignored.

File: .github/workflows/ci-nix.yml (hunk 1 above)
Comment (bradleystachurski) [reply .github/workflows/ci-nix.yml]: Verified `continue-on-error: true` will work on forks 👍 

This is a successful run on my fork's master branch: https://github.com/bradleystachurski/fedimint/actions/runs/6128318965

File: .github/workflows/daily.yml [hunk 2]
```diff
@@ -16,15 +16,14 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
 
   notifications:
-    # if: github.ref == 'refs/heads/master'
-    if: always()                              
+    if: always() && github.repository == 'fedimint/fedimint'
     name: "Notifications"
     timeout-minutes: 1
     runs-on: ubuntu-22.04
-    needs: [ check-unused-dependencies, audit ]
```
Comment (bradleystachurski) [inline .github/workflows/daily.yml:24]: We removed the `check-unused-dependencies` job in https://github.com/fedimint/fedimint/commit/b7a6e368192114376c1818f774f13dcde9803442.

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -21,7 +21,8 @@ jobs:
       - name: Build docs
         run: nix build -L .#debug.workspaceDocExport
 
-      - name: Deploy docs
+      - if: github.repository == 'fedimint/fedimint'
```
Comment (bradleystachurski) [inline .github/workflows/docs.yml:24]: Since this job `runs-on: ubuntu-22.04` instead of buildjet, we can allow it to run until the deploy step on forks.

**Comment by bradleystachurski** [comment]
> There are two reasons for why a build might not work:
> 
> * Secrets being unavailable
>   
>   * Often this should be fixable by running the cachix action read-only
> * Runners being unavailable because we use BuildJet and the fork isn't
>   
>   * We could supply alternative config, but GH runners are so slow that it's probably best to just run CI locally (easily possible due to nix) or by just opening a draft PR.

Thanks for the context!

The latest changes will prevent jobs on forks that
- Are configured to run with buildjet
  - Since we don't define an `idle_timeout`, the job will hang waiting for an available runner that will never be available on the fork (assuming no buildjet setup)
- Attempt to access secrets that don't fail gracefully (everything except cachix)

The goal for these changes is to prevent spam for failed CI runs on forks that have GH actions enabled. The draft PR and local `just test-ci-all` approaches make the alternate fork configs unnecessary.

**Comment by bradleystachurski** [comment]
> Since we don't define an idle_timeout, the job will hang waiting for an available runner that will never be available on the fork (assuming no buildjet setup)

Scratch that, looks like it times out after 1 day 😃 
https://github.com/bradleystachurski/fedimint/actions/runs/6122825276

**Review by elsirion — APPROVED** [review]

**Pull request merged**